components/Ethdriver/src/ethdriver.c

```diff
@@ -264,13 +276,7 @@ int client_rx(int *len)
         return -1;
     }
     int ret;
-    int id = client_get_sender_id();
-    client_t *client = NULL;
-    for (int i = 0; i < num_clients; i++) {
-        if (clients[i].client_id == id) {
-            client = &clients[i];
-        }
-    }
+    client_t *client = eth_get_client();
     assert(client);
```

Comment:
I wonder, if we factor this out, the `assert()` here and in other places might have to become a runtime check also? It's nitpicking, but we could still assert inside the check for development builds as this it not supposed to fail practically.

Reply:
are you suggesting assert -> ZF_LOGF_IF?

Reply:
> So everytime a lookup function is called, the caller should not consider it fatal if this fails, but report this. Might be a stale object then also that is just to be ignored. Or the calling contract must be, that this never fails - but the this should be clearly stated.

In this case, the client_get_sender_id() is required to return a value within [0, num_clients) that is obtained from the badge of the caller. If this is not true at runtime, then there was something wrong with the component generation at build time and isn't something that can be recovered at runtime. 

ZF_LOGF_IF would be better because the abort won't be compiled out under release builds.  Changing this isn't blocking this PR though.